pkg/analytics/client.go

```diff
@@ -37,98 +45,87 @@ const (
 const datadogLogLevel = "_logLevel"
 const datadogStatus = "status"
 
-func NewClient(properties map[string]interface{}) *Client {
+func NewClient() *Client {
 	local := GetOrCreateAnalyticsFile()
 
-	client := &Client{
-		Properties: properties,
-	}
+	client := &Client{}
+	client.universalProperties = make(map[string]any)
 
 	// These will get validated in AttachAuthorizations
-	client.UserId = local.Id
-	client.Properties["validated"] = false
+	client.userId = local.Id
+	client.universalProperties["validated"] = false
 
-	client.Properties["localId"] = local.Id
+	client.universalProperties["localId"] = local.Id
 	if runUuid, err := uuid.NewRandom(); err == nil {
-		client.Properties["runId"] = runUuid.String()
+		client.universalProperties["runId"] = runUuid.String()
 	}
 
 	return client
 }
 
 func (t *Client) AttachAuthorizations(claims *auth.KlothoClaims) {
-	t.Properties["localId"] = t.UserId
-	t.UserId = claims.Email
-	t.Properties["validated"] = claims.EmailVerified
+	t.universalProperties["localId"] = t.userId
+	t.userId = claims.Email
+	t.universalProperties["validated"] = claims.EmailVerified
 }
 
 func (t *Client) Info(event string) {
-	t.Properties[datadogLogLevel] = Info
-	t.track(event)
-}
-
-func (t *Client) Debug(event string) {
-	t.Properties[datadogLogLevel] = Debug
-	t.track(event)
+	t.Send(t.createPayload(Info, event))
 }
 
 func (t *Client) Warn(event string) {
-	t.Properties[datadogLogLevel] = Warn
-	t.Properties[datadogStatus] = Warn
-	t.track(event)
+	t.Send(t.createPayload(Warn, event))
 }
 
 func (t *Client) Error(event string) {
-	t.Properties[datadogLogLevel] = Error
-	t.Properties[datadogStatus] = Error
-	t.track(event)
+	t.Send(t.createPayload(Error, event))
 }
 
-func (t *Client) Panic(event string) {
-	t.Properties[datadogLogLevel] = Panic
-	// Using error since datadog does not support panic for the reserved status field
-	t.Properties[datadogStatus] = Error
-	t.track(event)
+func (p *Payload) addError(err error) *Payload {
+	p.Properties["error"] = fmt.Sprintf("%+v", err)
+	return p
 }
 
 func (t *Client) AppendProperties(properties map[string]interface{}) {
 	for k, v := range properties {
-		t.Properties[k] = v
+		t.universalProperties[k] = v
 	}
 }
 
-func (t *Client) DeleteProperty(key string) {
-	delete(t.Properties, key)
-}
-
 func (t *Client) UploadSource(source *core.InputFiles) {
 	data, err := CompressFiles(source)
 	if err != nil {
 		zap.S().Warnf("Failed to upload debug bundle. %v", err)
 		return
 	}
-	srcClient := &Client{
-		UserId:     t.UserId,
-		Properties: t.Properties,
-		Source:     data,
-	}
-	srcClient.Info("klotho uploading")
+	p := t.createPayload(Info, "klotho uploading")
+	p.Source = data
+	t.Send(p)
 }
 
-func (t *Client) track(event string) {
-	t.Event = event
-	err := SendTrackingToServer(t)
-
-	if err != nil {
-		zap.L().Debug(fmt.Sprintf("Failed to send metrics info. %v", err))
+func (t *Client) createPayload(level LogLevel, event string) *Payload {
```

Comment:
Does this need to return a pointer?

Reply:
Done mostly for efficiency (though it's not a big struct), and especially since this is an un-exported func.

Is there any sort of convention for when to return a pointer vs a plain struct?